cabal-install/Distribution/Client/RebuildMonad.hs

```diff
@@ -166,3 +176,58 @@ monitorDirectoryStatus dir = do
                     then monitorDirectory dir
                     else monitorNonExistentDirectory dir]
 
+doesFileExistMonitored :: FilePath -> Rebuild Bool
+doesFileExistMonitored f = do
+    root <- askRoot
+    exists <- liftIO $ doesFileExist (root </> f)
+    -- TODO: If the file exists, should we really monitor the entire
+    -- file?!
```

Comment:
In principle we'd take an arg, passed through `findFileMonitored` etc that says if we care about content or mere existence, and all the new uses of it would use the full content. So I wouldn't bother, but we can update the TODO here to say this.